xtask/src/commands/changeset/mod.rs

```diff
@@ -0,0 +1,499 @@
+///               ./matching_pull_request.graphql
+///
+mod matching_pull_request;
+mod scalars;
+
+use ::reqwest::Client;
+use anyhow::Result;
+use console::style;
+use dialoguer::{console::Term, theme::ColorfulTheme, Confirm, Editor, Input, Select};
+use git2;
+use itertools::Itertools;
+use matching_pull_request::matching_pull_request::Variables;
+use matching_pull_request::MatchingPullRequest;
+use memorable_wordlist;
+use serde::Serialize;
+use std::{fmt, fs, path::PathBuf, slice::Iter, str::FromStr};
+use structopt::StructOpt;
+use tinytemplate::{format_unescaped, TinyTemplate};
+use xtask::*;
+
+#[derive(Serialize)]
+struct TemplateResource {
+    number: String,
+    url: String,
+}
+
+#[derive(Serialize)]
+struct TemplateContext {
+    pulls: Vec<TemplateResource>,
+    issues: Vec<TemplateResource>,
+    title: String,
+    body: String,
+    author: String,
+}
+
+const REPO_WITH_OWNER: &'static str = "apollographql/router";
+
+const EXAMPLE_TEMPLATE: &'static str = "### {title} {{ for issue in issues -}}
+([Issue #{issue.number}]({issue.url})){{ if not @last }}, {{ endif }}
+{{- endfor }}
+
+{body}
+
+By [@{author}](https://github.com/{author}) in {{ for pull in pulls -}}
+{pull.url}{{ if not @last }}, {{ endif }}
+{{- endfor }}
+";
+
+impl Command {
+    pub fn run(&self) -> Result<()> {
+        match self {
+            Command::Create(command) => command.run(),
+        }
+    }
+}
+
+#[derive(Debug, StructOpt)]
+pub enum Command {
+    /// Add a new changeset
+    Create(Create),
+}
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash)]
+enum Classification {
+    Breaking,
+    Feature,
+    Fix,
+    Configuration,
+    Maintenance,
+    Documentation,
+    Experimental,
+}
+
+impl Classification {
+    /// These "short names" are the prefixes that are used on the files
+    /// themselves and also for the `--class` flag for the CLI.
+    fn as_short_name(&self) -> &'static str {
+        match self {
+            Classification::Breaking => "breaking",
+            Classification::Feature => "feat",
+            Classification::Fix => "fix",
+            Classification::Configuration => "config",
+            Classification::Maintenance => "maint",
+            Classification::Documentation => "docs",
+            Classification::Experimental => "exp",
+        }
+    }
+
+    /// This definition of how to iterate over classifications is responsible
+    /// for the ordering that eventually appears in the emitted CHANGELOG. It's
+    /// also the order options are displayed in the TUI.
+    pub fn iterator() -> Iter<'static, Classification> {
+        static CLASSIFICATIONS: [Classification; 7] = [
+            Classification::Breaking,
+            Classification::Feature,
+            Classification::Fix,
+            Classification::Configuration,
+            Classification::Maintenance,
+            Classification::Documentation,
+            Classification::Experimental,
+        ];
+        CLASSIFICATIONS.iter()
```

Comment:
~~nit: this returns an iterator over `&Classification` items, but it might be slightly more convenient for callers to iterate `Classification` items directly instead (since they’re `Copy`)~~

~~You can do that by replacing `.iter()` with `.into_iter()`. This changes the returned iterator type. To avoid naming it you can instead specify `impl Iterator<Item  = Classification>` as the return type. This means "some concrete but unnamed type, (almost) all the caller is allowed to assume is that this trait is implemented"~~

On further reading, maybe it shouldn’t be an iterator at all. See below.